flink/src/test/java/org/apache/iceberg/flink/TestFlinkFilters.java

```diff
@@ -0,0 +1,361 @@
+    Assert.assertTrue(actual.isPresent());
+    UnboundPredicate<Object> expected = org.apache.iceberg.expressions.Expressions.isNull("field1");
+    assertPredicatesMatch(expected, (UnboundPredicate<?>) actual.get());
+  }
+
+  @Test
+  public void testIsNotNull() {
+    Expression expr = resolve(Expressions.$("field1").isNotNull());
+    Optional<org.apache.iceberg.expressions.Expression> actual = FlinkFilters.convert(expr);
+    Assert.assertTrue(actual.isPresent());
+    UnboundPredicate<Object> expected = org.apache.iceberg.expressions.Expressions.notNull("field1");
+    assertPredicatesMatch(expected, (UnboundPredicate<?>) actual.get());
+  }
+
+  @Test
+  public void testAnd() {
+    Expression expr = resolve(
+        Expressions.$("field1").isEqual(Expressions.lit(1)).and(Expressions.$("field2").isEqual(Expressions.lit(2L))));
+    Optional<org.apache.iceberg.expressions.Expression> actual = FlinkFilters.convert(expr);
+    Assert.assertTrue(actual.isPresent());
+    And and = (And) actual.get();
+    And expected = (And) org.apache.iceberg.expressions.Expressions.and(
+        org.apache.iceberg.expressions.Expressions.equal("field1", 1),
+        org.apache.iceberg.expressions.Expressions.equal("field2", 2L));
+
+    Assert.assertEquals(expected.op(), and.op());
+    Assert.assertEquals(expected.left().op(), and.left().op());
+    Assert.assertEquals(expected.right().op(), and.right().op());
+  }
+
+  @Test
+  public void testOr() {
+    Expression expr = resolve(
+        Expressions.$("field1").isEqual(Expressions.lit(1)).or(Expressions.$("field2").isEqual(Expressions.lit(2L))));
+    Optional<org.apache.iceberg.expressions.Expression> actual = FlinkFilters.convert(expr);
+    Assert.assertTrue(actual.isPresent());
+    Or or = (Or) actual.get();
+    Or expected = (Or) org.apache.iceberg.expressions.Expressions.or(
+        org.apache.iceberg.expressions.Expressions.equal("field1", 1),
+        org.apache.iceberg.expressions.Expressions.equal("field2", 2L));
+
+    Assert.assertEquals(expected.op(), or.op());
+    Assert.assertEquals(expected.left().op(), or.left().op());
+    Assert.assertEquals(expected.right().op(), or.right().op());
+  }
+
+  @Test
+  public void testNot() {
+    Expression expr = resolve(ApiExpressionUtils.unresolvedCall(
+        BuiltInFunctionDefinitions.NOT, Expressions.$("field1").isEqual(Expressions.lit(1))));
+    Optional<org.apache.iceberg.expressions.Expression> actual = FlinkFilters.convert(expr);
+    Assert.assertTrue(actual.isPresent());
+    Not not = (Not) actual.get();
+    Not expected = (Not) org.apache.iceberg.expressions.Expressions.not(
+        org.apache.iceberg.expressions.Expressions.equal("field1", 1));
+
+    Assert.assertEquals(expected.op(), not.op());
+    assertPredicatesMatch((UnboundPredicate<?>) expected.child(), (UnboundPredicate<?>) not.child());
+  }
+
+  @Test
+  public void testLike() {
```

Comment:
This suite also needs negative tests for `LIKE` with different wildcard locations:
* The pattern is `"%abc"`
* The pattern is `"%abc%"`
* The pattern is `"abc%d"`
* The pattern is `"%"`

Reply:
Also, could you test the wildcard behavior with `_`?

Reply:
I add the test case ([link](https://github.com/zhangjun0x01/iceberg/blob/filterPushDown/flink/src/test/java/org/apache/iceberg/flink/TestFlinkFilters.java#L300)) ,and add the sql test case ([link](https://github.com/zhangjun0x01/iceberg/blob/filterPushDown/flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java#L580))